### PR TITLE
Crpl 2905 add documentation for the used

### PR DESCRIPTION
**Pull request by csabee1979**

### CODE REVIEW

File: html/xpath value finder (internal)/35629a5f-aad8-4c08-8946-7a46af449595.json [hunk 1]
```diff
@@ -18,25 +18,30 @@
...
-        "6c326055-5409-47ab-8197-305240922f8e": "const xpath = require(\"xpath\");\nconst parse5 = require(\"parse5\");\nconst xmlser = require(\"xmlserializer\");\nconst dom = require(\"xmldom\").DOMParser;\n\nconst { document, query } = data;\nconst doc = new dom({ \n  errorHandler: {\n    warning: () => {},\n    error: console.error\n  }\n}).parseFromString(document);\n\nconst select = xpath.useNamespaces({ \"x\": \"http://www.w3.org/1999/xhtml\" });\nconst nodes = select(query, doc);\n\nconst values = nodes.map((node) => node.toString());\n\noutputs.results(values, tag)"
+        "6c326055-5409-47ab-8197-305240922f8e": "const xpath = require(\"xpath\");\nconst parse5 = require(\"parse5\");\nconst xmlser = require(\"xmlserializer\");\nconst dom = require(\"xmldom\").DOMParser;\n\nconst { document, query } = data;\nconst doc = new dom({ \n  errorHandler: {\n    warning: () => {},\n    error: (e) => {outputs.error(e.toString(), tag)}\n  }\n}).parseFromString(document);\n\nconst select = xpath.useNamespaces({ \"x\": \"http://www.w3.org/1999/xhtml\" });\nconst nodes = select(query, doc);\n\nconst values = nodes.map((node) => node.toString());\n\noutputs.results(values, tag)"
```
Comment (lvass74) [inline html/xpath value finder (internal)/35629a5f-aad8-4c08-8946-7a46af449595.json:21]: I think error output should be at least of type `{error: string}`

File: html/xpath value finder/4ea708d0-8092-441e-bbd5-ef333d52e1dc.json [hunk 2]
```diff
@@ -60,23 +65,35 @@
...
       "a2deb91f-4903-4e5b-b4fa-235e713d5cd8": {
         "type": "string[]",
         "name": "values",
-        "description": "Values of the nodes matching the specified query"
+        "description": "Values of the nodes matching the specified query.\n\nExample:\n\n[\"Content\"]"
+      },
+      "39868688-9f5a-44f9-af18-6a624918b71a": {
+        "type": "{\"error\": string}",
+        "name": "error",
+        "description": "Sends the error, if an"
```
Comment (lvass74) [inline html/xpath value finder/4ea708d0-8092-441e-bbd5-ef333d52e1dc.json:85]: Typo: "if any"

File: string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json [hunk 3]
```diff
@@ -5,21 +5,25 @@
...
         "type": "string",
-        "name": "string"
+        "name": "string",
+        "description": "Receives the string to be replaced.\n\nExample:\n\n\"dog cat rain dog\""
       },
       "d1118ac0-749b-46e9-81af-8fc74dc14e10": {
         "type": "string",
-        "name": "regex"
+        "name": "regex",
+        "description": "Receives regular expression to test the specified string against.\n\nExample: \n\n\"dog\""
```
Comment (lvass74) [inline string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json:14]: I'd rather a more RegExp like example here. "dog" seems like a normal string. What about something like these:
`[Dd]og` or `d.g` or ` *dog *`?

File: string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json [hunk 4]
```diff
@@ -5,21 +5,25 @@
...
       },
       "3490f391-9b3b-441e-b4a2-b078669cf5b4": {
         "type": "string",
-        "name": "new"
+        "name": "new",
+        "description": "Receives the replacement string.\n\nExample:\n\n\"cat\""
```
Comment (lvass74) [inline string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json:19]: Can we use back references here? If yes, we should have an example for.

File: string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json [hunk 5]
```diff
@@ -1,28 +1,38 @@
 {
-  "description": "Executes a global regex pattern match.\n\nFlavour depends on the running environment.",
+  "description": "Matches a string to a non-global regular expression, and returns substrings that match the regular expression's groups.\n\n",
```
Comment (lvass74) [inline string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json:2]: Actually it not only returns the groups and only returns them if they are present in the regexp so I would rephrase the description or keep it simple

File: string/regex matcher/internal/70ea8986-7c87-49bb-aa42-c2d1b970c7d4.json [hunk 6]
```diff
@@ -12,13 +12,17 @@
         "type": "boolean",
         "name": "matches"
       },
+      "dd49cf66-19bb-4546-8f30-ccc742b66d73": {
+        "type": "string",
+        "name": "matched"
+      },
       "7866a8a3-a295-4bed-832f-e5c901996c42": {
         "type": "any",
```
Comment (lvass74) [inline string/regex matcher/internal/70ea8986-7c87-49bb-aa42-c2d1b970c7d4.json:20]: It is rather `string[]`

File: string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json [hunk 7]
```diff
@@ -1,28 +1,38 @@
...
+        "description": "Sends the matched text.\n\nExample: \n\n\"fox jumps over\""
       },
       "0ae97343-7bd6-43c7-bc0a-08acc2b964c8": {
         "type": "string[]",
         "name": "groups",
-        "description": "Groups matched by the given regex pattern"
+        "description": "Sends the capturing groups matched by the given regex pattern.\n\nExample:\n```\n[ \"fox\" ]\n```"
+      },
+      "856b46c6-6d81-4023-989e-3e312f051e02": {
+        "type": "({string: string} or null)",
```
Comment (lvass74) [inline string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json:33]: I think we decided to emit empty object instead of null it there weren't any named groups, didn't we?

**Review by lvass74 — CHANGES_REQUESTED** [review]

File: html/xpath value finder (internal)/35629a5f-aad8-4c08-8946-7a46af449595.json (hunk 1 above)
Comment (csabee1979) [reply html/xpath value finder (internal)/35629a5f-aad8-4c08-8946-7a46af449595.json:21]: Good catch!

File: string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json (hunk 3 above)
Comment (csabee1979) [reply string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json:14]: Good idea!

File: string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json (hunk 7 above)
Comment (csabee1979) [reply string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json:33]: You are right.

File: string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json (hunk 4 above)
Comment (csabee1979) [reply string/global regex replacer/5719fd66-d597-45ee-bff8-64106b48879e.json:19]: I don't know wheather it is a good idea. Isn't it a javascript feature?
